tests/test_worker_extractor.py

```diff
@@ -133,92 +185,75 @@ def test_extraction_from_html(self):
         contents = "This is a test file. I'm testing PyPLN extractor worker!"
         filename = os.path.join(DATA_DIR, 'text_file')
         data = {'filename': filename,
-                'contents': base64.b64encode(contents)}
-        doc_id = self.collection.insert(data, w=1)
-        Extractor().delay(doc_id)
-        refreshed_document = self.collection.find_one({'_id': doc_id})
-        self.assertEqual(refreshed_document['mimetype'], 'text/plain')
+                'contents': base64.b64encode(contents.encode('utf-8'))}
+        result = Extractor().process(data)
+        self.assertEqual(result['mimetype'], 'text/plain')
 
     def test_unknown_mimetype_should_be_flagged(self):
         filename = os.path.join(DATA_DIR, 'random_file')
         # we can't put the expected text content here, so we'll just make sure
         # it's equal to the input content, since
-        contents = open(filename).read()
-        data = {'filename': filename,
-                'contents': base64.b64encode(contents)}
-        doc_id = self.collection.insert(data, w=1)
-        Extractor().delay(doc_id)
-        refreshed_document = self.collection.find_one({'_id': doc_id})
-        self.assertEqual(refreshed_document['mimetype'], 'unknown')
-        self.assertEqual(refreshed_document['text'], "")
-        self.assertEqual(refreshed_document['language'], "")
-        self.assertEqual(refreshed_document['file_metadata'], {})
+        contents = open(filename, 'rb').read()
+        result = Extractor().process({'filename': filename,
+                                      'contents': base64.b64encode(contents)})
+        self.assertEqual(result['mimetype'], 'unknown')
+        self.assertEqual(result['text'], "")
+        self.assertEqual(result['language'], "")
+        self.assertEqual(result['file_metadata'], {})
 
     def test_unknown_encoding_should_be_ignored(self):
         filename = os.path.join(DATA_DIR, 'encoding_unknown_to_libmagic.txt')
-        expected = u"This file has a weird byte (\x96) that makes it impossible for libmagic to recognize it's encoding."
+        expected = "This file has a weird byte (\x96) that makes it " \
+                   "impossible for libmagic to recognize it's encoding."
         data = {'filename': filename,
-                'contents': base64.b64encode(open(filename).read())}
-        doc_id = self.collection.insert(data, w=1)
-        Extractor().delay(doc_id)
-        refreshed_document = self.collection.find_one({'_id': doc_id})
-        self.assertEqual(refreshed_document['text'], expected)
-        self.assertEqual(refreshed_document['file_metadata'], {})
-        self.assertEqual(refreshed_document['language'], 'en')
+                'contents': base64.b64encode(open(filename, 'rb').read())}
+        result = Extractor().process(data)
+        self.assertEqual(result['text'], expected)
+        self.assertEqual(result['file_metadata'], {})
+        self.assertEqual(result['language'], 'en')
```

Comment:
This is the test that is still not passing because pycld2 is blowing up with the weird byte, right? 

I don't know exactly what to do about this. I don't think any of the options is very good. From what I remembered you gave me three options (of course we can think of new ones):

1. Try to use the old cld version
2. Not detect language for this documents
3. Try to remove the weird byte somehow

`1` sounds terrible because it is old, unmaintained code, that does not support python3. It's my last option!

`2` has one big problem: if we don't detect language, we won't run palavras for the documents that are in Portuguese. This means a lot of the important analysis won't happen.

`3` sounds a bit more interesting if we can do it the right way. I'm worried about two things: we should only try to do it when we detect the weird byte in the document, and it also should be reasonably fast. This options has another limitation, though: if we find different, new weird bytes, we have to change our code to check for them also.

@fccoelho Any ideas on how to deal with this?